platform/mellanox/install-pending-fw.py

```diff
@@ -0,0 +1,53 @@
+#!/usr/bin/env python3
+
+from fwutil.lib import ComponentStatusProvider, PlatformComponentsParser
+from sonic_platform.component import ComponentCPLD
+
+#Globals
+CPLD_FLAG = None
+
+# Init platform chassis helper classes
+csp = ComponentStatusProvider()
+pcp = PlatformComponentsParser(csp.is_modular_chassis())
+
+
+# Parse update status file
+update_status = csp.read_au_status_file_if_exists()
+
+if update_status is None:
+    exit(0)
+
+# Parse platform components file
+pcp.parse_platform_components()
```

Comment:
@alexrallen how do you expect to handle errors here?

Reply:
I can bail out of the script with an error message if it throws an exception. Is there an exception we particularly need to catch?

Reply:
@alexrallen so the idea is to deliver a raw trace back to the user?

Reply:
Not a raw trace, I will catch the error and output a message to the log. The question is what error message do you think I should give and is there a particular python exception I should catch?

i.e. do we anticipate this failing in a specific way? Just an ill-formated `platform_components.json` file?

Reply:
> i.e. do we anticipate this failing in a specific way? Just an ill-formated platform_components.json file?
@alexrallen why not?

Reply:
Ok I will catch a broad Exception and then report back that parsing of the platform_components.json file failed and will also report the error message from python. Sound good?

The script will also bail at this point. I will add a sleep so someone can catch the error before reboot.

Reply:
@alexrallen ok